src/main/java/io/github/thebusybiscuit/slimefun4/implementation/listeners/AncientAltarListener.java

```diff
@@ -156,8 +156,18 @@ private void usePedestal(@Nonnull Block pedestal, @Nonnull Player p) {
             Slimefun.runSync(() -> removedItems.remove(uuid), 30L);
 
             entity.remove();
-            p.getInventory().addItem(pedestalItem.getOriginalItemStack(entity));
             p.playSound(pedestal.getLocation(), Sound.ENTITY_ITEM_PICKUP, 1F, 1F);
+
+            /*
+             * Fixes #3476
+             * Drop the item instead if the player's
+             * inventory is full
+             */
+            if (p.getInventory().firstEmpty() != -1) {
```

Comment:
This will not check if the item actually fits in the target inventory. Best to add the items but then check the returned Map<Integer, ItemStack> and drop what if left behind. This way item retrieval will work with a full inventory but with space in a stack for the item being removed

Reply:
It does add stacked items to another stacked ones but does this imply It must drop the remaining items that cannot be stacked onto other items anymore?

Reply:
The pedestal only accepts one item at a time though and do not support placing stacked items, I could just get the returned map assign it to a variable and for each value is gonna be dropped

Reply:
Very well, it makes it much simple and since the pedestal only supports one amount of  placed item at a time and it can only return a single amount item